Source/events.filtering/FilterEventCallback.ts

```diff
@@ -3,4 +3,4 @@
 
 import { EventContext } from '@dolittle/sdk.events';
 
-export type FilterEventCallback = (event: any, context: EventContext) => boolean;
+export type FilterEventCallback = (event: any, context: EventContext) => boolean | Promise<boolean>;
```

Comment:
Isn't this a bit hard to work with if it can return both boolean and Promise<boolean>? Wouldn't it be better to just return Promise<boolean>? Since as it is now the result of this callback should be awaited every time since we don't know if it returns the boolean or a Promise. And from the implementors stand point if there isn't any asynchnosity in the FilterEventCallback it could just resolve to the value

Reply:
Nope, it's very easy. If you `await` something that is not a promise, it resolves immediately. So no need to force people to return Promises if they don't want to.